### PR TITLE
corrects exposing of inline definitions

### PR DESCRIPTION
**Pull request by LeFnord**
solves:

- https://github.com/ruby-grape/grape-swagger/pull/503
- https://github.com/ruby-grape/grape-swagger-entity/issues/13

### CODE REVIEW

**Comment by LeFnord** [comment]
solves:

- https://github.com/ruby-grape/grape-swagger/pull/503
- https://github.com/ruby-grape/grape-swagger-entity/issues/13

File: lib/grape-swagger/endpoint.rb [hunk 1]
```diff
@@ -288,7 +288,8 @@ def expose_params_from_model(model)
     end
 
     def model_name(name)
-      name.respond_to?(:name) ? name.name.demodulize.camelize : name.split('::').last
+      name = name.to_s.split('::')[0..-2].join('::').constantize if name.to_s.end_with?('Entity', 'Entities')
```
Comment (Bugagazavr) [inline lib/grape-swagger/endpoint.rb]: Why we can't just drop all '::', and show model name as is?

Comment (LeFnord) [inline lib/grape-swagger/endpoint.rb]: good point … I did it, cause the name could be very long, see the spec, without it the name would be: `TestDefinitionClass1`

Comment (LeFnord) [inline lib/grape-swagger/endpoint.rb]: @Bugagazavr … what is your meaning?

Comment (Bugagazavr) [inline lib/grape-swagger/endpoint.rb]: i think we can trim symbols if maximum defenition length is reached

Comment (Bugagazavr) [inline lib/grape-swagger/endpoint.rb]: `name.to_s.split('').reverse.take(255).reverse.join`

Comment (LeFnord) [inline lib/grape-swagger/endpoint.rb]: ok, I'll chang it 😄

Comment (Bugagazavr) [inline lib/grape-swagger/endpoint.rb]: also we can add `respond_to?(:entity_name)` to customize name manually

Comment (LeFnord) [inline lib/grape-swagger/endpoint.rb]: where would it it be set?

Comment (Bugagazavr) [inline lib/grape-swagger/endpoint.rb]: i thin this should be placed in `model_name` method

```rb
if name.respond_to?(:entity_name)
  name.entity_name
else
  name.to_s.gsub('::', '').split('').reverse.take(255).reverse.join
end
```

Comment (LeFnord) [inline lib/grape-swagger/endpoint.rb]: yeap, but where in the entry/API definition would it be set?

File: lib/grape-swagger/endpoint.rb [hunk 2]
```diff
@@ -288,7 +288,17 @@ def expose_params_from_model(model)
...
+      if name.respond_to?(:entity_name)
+        name.entity_name
+      elsif name.to_s.end_with?('Entity', 'Entities')
+        length = 0
+        name.to_s.split('::')[0..-2].reverse.take_while do |x|
+          length += x.length
+          length < 42
+        end.reverse.join
+      else
+        name.name.demodulize.camelize
```
Comment (LeFnord) [inline lib/grape-swagger/endpoint.rb]: I let it, cause without it, it would be a breaking change

**Comment by serggl** [comment]
I guess this is a late comment, but why there is 2 different code blocks for parsing model name?

- One that was just updated in this PR: https://github.com/ruby-grape/grape-swagger/blob/master/lib/grape-swagger/endpoint.rb#L290

- and another one here: https://github.com/ruby-grape/grape-swagger/blob/master/lib/grape-swagger/doc_methods/data_type.rb#L43

Second question is what is the magic logic of dropping `Entity` tail in these methods? If I comment this out - specs keep passing...that is confusing

**Comment by LeFnord** [comment]
@serggl … thanks for spotting on it … it seems you have the knowledge to refatore it out,
please can you make a PR, thanks

**Comment by serggl** [comment]
@LeFnord sure. On the way

**Pull request merged**